### PR TITLE
Add basic backend code

### PR DESCRIPTION
**Pull request by Samyakkumar**
Added code to the frontend so that it calls the studentRouter and the studentInfo router on the parent portal page and actually gets the data from the backend.

Added a fix so that while sending a POST request to the student router, it saves it and can be accessed later on.

Screen Shot 2019-03-11 at 12 57 43 PM

It shows port 3001 on the screenshot, but it is using 3000 in the code(Which is what it should be) . That is some issue with my computer.

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "client",
   "version": "0.1.0",
-  "proxy": "http://localhost:3001/",
+  "proxy": "http://localhost:3000",
```
Comment (SujeethJinesh) [inline client/package.json]: Is this related to the issue you were having? Is there a specific reason we're changing it to port 3001?

File: client/src/components/Announcements.js [hunk 2]
```diff
@@ -1,6 +1,10 @@
 import React, { Component } from 'react';
 import Announcement from './Announcement'
 import './css/Announcements.css'
+
+const API_ANNOUNCEMENTS_DEV = "http://localhost:5000/api/schoolInfo/announcements"
+const API_HOUSEHOLD_PROD = "";
```
Comment (SujeethJinesh) [inline client/src/components/Announcements.js]: Add a TODO comment here to make a production endpoint

File: client/package.json (hunk 1 above)
Comment (Samyakkumar) [reply client/package.json]: We're still going to continue using 3000. It's just that for my machine, I need to use 3001 until I figure out what is using 3000

File: client/src/components/Announcements.js (hunk 2 above)
Comment (SujeethJinesh) [reply client/src/components/Announcements.js]: Also I would keep the naming more consistent. `API_ANNOUNCEMENTS_PROD`

File: client/src/components/Announcements.js [hunk 3]
```diff
@@ -10,13 +14,61 @@ class AnnouncementList extends Component {
...
+    }
+
+    /**
+     * Makes a GET request to either dev or prod and fetches the announcement list
+     * as soon as the component loads and updates the state
+     * TODO: Add Validation for output
+     * TODO: Once finished, remove debugging statements
+     */
+    componentDidMount() {
+        if (this.ReactIsInDevelomentMode()) {
```
Comment (SujeethJinesh) [inline client/src/components/Announcements.js]: The only difference between these two conditional statements is `urlToFetch`, I would simplify it to:

```
var urlToFetch = API_ANNOUNCEMENTS_PROD
if (this.ReactIsInDevelomentMode()) {
    urlToFetch = API_ANNOUNCEMENTS_DEV;
}
fetch(urlToFetch)
            .then(response => response.json())
            .then(data =>  this.setState({announcements:this.parseAnnouncements(data)}))
```

File: client/package.json (hunk 1 above)
Comment (SujeethJinesh) [reply client/package.json]: Ah gotcha.

File: client/src/components/Announcements.js [hunk 4]
```diff
@@ -10,13 +15,61 @@ class AnnouncementList extends Component {
...
+          } else {
+            let urlToFetch = API_ANNOUNCEMENTS_DEV;
+            fetch(urlToFetch)
+            .then(response => response.json())
+            .then(data =>  this.setState({announcements:this.parseAnnouncements(data)}))
+          }
     }
 
+    /**
+     * TODO: How are multiple announcements stored? in data.annoucnment or just as a string
```
Comment (SujeethJinesh) [inline client/src/components/Announcements.js]: Here's what it looks like:
https://github.com/GTBitsOfGood/bright-futures-academy/pull/83

Let me know if you need some help with making the calls

File: client/src/components/Announcements.js [hunk 5]
```diff
@@ -10,11 +15,54 @@ class AnnouncementList extends Component {
...
+     * Assuming the object has an attribute announcement for now
+     * 
+     * Returns formatted data to be stored in the state
+     */
+    parseAnnouncements(data) {
+        if (data[0].default !== undefined) {
```
Comment (SujeethJinesh) [inline client/src/components/Announcements.js]: The condition for this should be switched. You want to check if there is an announcements key first and if it's not there, then you should use default.

File: client/src/components/Announcements.js [hunk 6]
```diff
@@ -10,11 +15,54 @@ class AnnouncementList extends Component {
...
+                toBeReturned.push(data[i].announcement)
+            }
+          return toBeReturned;
+        }
+      } 
+
+      /**
+       * Return whether React is in dev or production
+       */
+    ReactIsInDevelomentMode(){
```
Comment (SujeethJinesh) [inline client/src/components/Announcements.js]: This method is probably going to be used in a lot of places. I would bust it out into a utils file and import it where you need it.

File: client/src/pages/ParentPortal.js [hunk 7]
```diff
@@ -13,29 +18,52 @@ class PaymentPortal extends Component {
...
+      .then(data => this.setState({students: data.students}))
+    }
   }
 
-  //TODO: ADD KEYS FOR ANNOUNCEMENTS. 
-  //TODO: Add logic for page changing. 
+  /**
+  * Return whether React is in dev or production
+  */
+  ReactIsInDevelomentMode(){
```
Comment (SujeethJinesh) [inline client/src/pages/ParentPortal.js]: Mentioned before, move to utils file

File: client/src/pages/ParentPortal.js [hunk 8]
```diff
@@ -13,29 +18,52 @@ class PaymentPortal extends Component {
...
+  /**
+   * Makes a GET request to either dev or prod and fetches the student list
+   * as soon as the component mounts and updates the state
+   */
+  componentDidMount(){
+    //TODO: Figure out how it is being passed later : this.props.houseHoldID
+    //TODO: Add logic for retrieving householdID
+    if (this.ReactIsInDevelomentMode()){
+      //Fetch the student list
+      let urlToFetch = API_STUDENT_DEV + '5c8680ffad46ec4f26e7b46f';
```
Comment (SujeethJinesh) [inline client/src/pages/ParentPortal.js]: Same as comment as in announcements, simplify by having urlToFetch be the only thing being changed. Also what is "5c8680ffad46ec4f26e7b46f"?

File: client/src/pages/ParentPortal.js [hunk 9]
```diff
@@ -13,29 +18,52 @@ class PaymentPortal extends Component {
...
+  /**
+  * Return whether React is in dev or production
+  */
+  ReactIsInDevelomentMode(){ 
+    return '_self' in React.createElement('div');
+  }
+ 
   //TODO: Add logic for making payment
   render() {
+    console.log("Rendering")
```
Comment (SujeethJinesh) [inline client/src/pages/ParentPortal.js]: Let's not log stuff on the front end

**Comment by SujeethJinesh** [comment]
I forgot what we said about package-lock.json. Should we revert those files because prod will rebuild it anyway from package.json?

**Comment by SujeethJinesh** [comment]
Left some comments. You also have some conflicts, so I'd resolve those too. nice job! 🕺

File: client/src/pages/ParentPortal.js (hunk 8 above)
Comment (spencerhyphengold) [reply client/src/pages/ParentPortal.js]: It should be the household id for a dummy entry. Add a TODO to replace this with household_id as pulled from Redux.

File: client/src/pages/ParentPortal.js [hunk 10]
```diff
@@ -13,29 +18,52 @@ class PaymentPortal extends Component {
...
       <PrimaryNavBar />
       <div className="container"> 
-      <StudentCard studentName={"Name1"} studentID={1} studentBalance={1234}/>
-          <StudentCard studentName={"Name2"} studentID={2} studentBalance={123}/>
-          <StudentCard studentName={"Name2"} studentID={2} studentBalance={123}/>
-          <StudentCard studentName={"Name2"} studentID={2} studentBalance={123}/>
-          <StudentCard studentName={"Name2"} studentID={2} studentBalance={123}/>
-          <StudentCard studentName={"Name2"} studentID={2} studentBalance={123}/>
-      <AnnouncementList announcements={["some stuff1", "some stuff2", "some stuff3"]}/>
+          {this.state.students.map(function(d, id) {
```
Comment (spencerhyphengold) [inline client/src/pages/ParentPortal.js]: Is there any reason to include id as a param here?

File: client/src/pages/ParentPortal.js [hunk 11]
```diff
@@ -13,29 +19,48 @@ class PaymentPortal extends Component {
...
+    .then(data => this.setState({students: data}))
+    
   }
 
-  //TODO: ADD KEYS FOR ANNOUNCEMENTS. 
-  //TODO: Add logic for page changing. 
+  /**
+  * Return whether React is in dev or production
+  */
+  ReactIsInDevelomentMode(){
```
Comment (SujeethJinesh) [inline client/src/pages/ParentPortal.js]: You can remove this now right?
